actor/v7action/application_summary_test.go

```diff
@@ -574,6 +573,115 @@ var _ = Describe("Application Summary Actions", func() {
 						)
 					})
 
+					When("getting application deployment succeeds", func() {
+						When("the deployment is active", func() {
+							When("the deployment strategy is rolling", func() {
+								When("the deployment is in progress", func() {
+									BeforeEach(func() {
+										fakeCloudControllerClient.GetDeploymentsReturns(
+											[]resources.Deployment{
+												{
+													GUID:         "some-deployment-guid",
+													Strategy:     "rolling",
+													StatusValue:  "ACTIVE",
+													StatusReason: "DEPLOYING",
+												},
+											},
+											nil,
+											nil,
+										)
+									})
+									It("returns the deployment information", func() {
+										Expect(summary.Deployment).To(Equal(resources.Deployment{
+											GUID:         "some-deployment-guid",
+											Strategy:     "rolling",
+											StatusValue:  "ACTIVE",
+											StatusReason: "DEPLOYING",
+										}))
+									})
+								})
+
+								When("the deployment is canceled", func() {
+									When("the deployment is in progress", func() {
+										BeforeEach(func() {
+											fakeCloudControllerClient.GetDeploymentsReturns(
+												[]resources.Deployment{
+													{
+														GUID:         "some-deployment-guid",
+														Strategy:     "rolling",
+														StatusValue:  "ACTIVE",
+														StatusReason: "CANCELLING",
+													},
+												},
+												nil,
+												nil,
+											)
+										})
+										It("returns the deployment information", func() {
+											Expect(summary.Deployment).To(Equal(resources.Deployment{
+												GUID:         "some-deployment-guid",
+												Strategy:     "rolling",
+												StatusValue:  "ACTIVE",
+												StatusReason: "CANCELLING",
+											}))
+										})
+									})
+								})
+							})
+						})
+
+						When("the deployment is finalized", func() {
+							BeforeEach(func() {
+								fakeCloudControllerClient.GetDeploymentsReturns(
+									[]resources.Deployment{
```

Comment:
A finalized deployment is not active. The code only retrieves active deployments. Hence the `resources.Deployment` will be blank.

Reply:
Perhaps this could be something like `when GetLatestActiveDeploymentForApp returns no deployments` to be more expressive?

Reply:
Since its BDD, won't it be more prudent not to use function names?

Reply:
Suggest `when there are no active deployments` maybe?

Also it's still confusing to me… why does `GetDeployment` return an *empty* `Deployment` and not a *null* `Deployment` in this case?

Reply:
Fixed the message

Reply:
And that function [does in fact return an _empty_ `Deployment` instead of a `null` one](https://github.com/cloudfoundry/cli/blob/9b35b376656308d8f2bcbf1251978c73b840c65d/actor/v7action/deployment.go#L32). Wild.

Ok, please consider this thread resolved.